File: update_home.py
import sys

file_path = "src/components/landing/Home.tsx"
with open(file_path, "r") as f:
    content = f.read()

# Update signature
content = content.replace(
    "export function Home({ onLogin, onRegisterOrg }: { onLogin: () => void, onRegisterOrg: () => void }) {",
    "export function Home({ onLogin, onRegisterOrg, onRegisterUser }: { onLogin: () => void, onRegisterOrg: () => void, onRegisterUser: () => void }) {"
)

# Update nav buttons
old_nav = """            <div style={{display:'flex', gap:'12px', alignItems:'center', marginLeft:'12px'}}>
              <button onClick={onLogin} style={{background: 'transparent', color: '#1e293b', border: '1px solid #cbd5e1', padding: '8px 16px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '14px'}}>Log In</button>
              <button onClick={onRegisterOrg} style={{background: 'var(--green)', color: '#fff', border: 'none', padding: '8px 16px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '14px'}}>Create Network</button>
            </div>"""
new_nav = """            <div style={{display:'flex', gap:'12px', alignItems:'center', marginLeft:'12px'}}>
              <button onClick={onLogin} style={{background: 'transparent', color: '#1e293b', border: '1px solid #cbd5e1', padding: '8px 16px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '14px'}}>Log In</button>
              <button onClick={onRegisterUser} style={{background: 'var(--green)', color: '#fff', border: 'none', padding: '8px 16px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '14px'}}>Create New Account</button>
            </div>"""
content = content.replace(old_nav, new_nav)

# Update mobile nav buttons
old_mobile_nav = """          <div style={{display:'flex', flexDirection:'column', gap:'8px', marginTop:'8px', paddingTop:'8px', borderTop:'1px solid var(--border)'}}>
            <button onClick={() => { setMobileMenuOpen(false); onLogin(); }} style={{background: 'transparent', color: '#1e293b', border: '1px solid #cbd5e1', padding: '12px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '15px'}}>Log In</button>
            <button onClick={() => { setMobileMenuOpen(false); onRegisterOrg(); }} style={{background: 'var(--green)', color: '#fff', border: 'none', padding: '12px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '15px'}}>Create Network</button>
          </div>"""
new_mobile_nav = """          <div style={{display:'flex', flexDirection:'column', gap:'8px', marginTop:'8px', paddingTop:'8px', borderTop:'1px solid var(--border)'}}>
            <button onClick={() => { setMobileMenuOpen(false); onLogin(); }} style={{background: 'transparent', color: '#1e293b', border: '1px solid #cbd5e1', padding: '12px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '15px'}}>Log In</button>
            <button onClick={() => { setMobileMenuOpen(false); onRegisterUser(); }} style={{background: 'var(--green)', color: '#fff', border: 'none', padding: '12px', borderRadius: '8px', cursor: 'pointer', fontWeight: 600, fontSize: '15px'}}>Create New Account</button>
          </div>"""
content = content.replace(old_mobile_nav, new_mobile_nav)

# Update Hero buttons
old_hero_btns = """            <div className="hero-btns">
              <a href="https://wa.me/27739441222?text=Hi+I+want+to+get+SafetyLink" target="_blank" rel="noreferrer" className="btn-wa">
                <svg width="18" height="18" viewBox="0 0 24 24" fill="white"><path d="M17.472 14.382c-.297-.149-1.758-.867-2.03-.967-.273-.099-.471-.148-.67.15-.197.297-.767.966-.94 1.164-.173.199-.347.223-.644.075-.297-.15-1.255-.463-2.39-1.475-.883-.788-1.48-1.761-1.653-2.059-.173-.297-.018-.458.13-.606.134-.133.298-.347.446-.52.149-.174.198-.298.298-.497.099-.198.05-.371-.025-.52-.075-.149-.669-1.612-.916-2.207-.242-.579-.487-.5-.669-.51-.173-.008-.371-.01-.57-.01-.198 0-.52.074-.792.372-.272.297-1.04 1.016-1.04 2.479 0 1.462 1.065 2.875 1.213 3.074.149.198 2.096 3.2 5.077 4.487.709.306 1.262.489 1.694.625.712.227 1.36.195 1.871.118.571-.085 1.758-.719 2.006-1.413.248-.694.248-1.289.173-1.413-.074-.124-.272-.198-.57-.347m-5.421 7.403h-.004a9.87 9.87 0 01-5.031-1.378l-.361-.214-3.741.982.998-3.648-.235-.374a9.86 9.86 0 01-1.51-5.26c.001-5.45 4.436-9.884 9.888-9.884 2.64 0 5.122 1.03 6.988 2.898a9.825 9.825 0 012.893 6.994c-.003 5.45-4.437 9.884-9.885 9.884m8.413-18.297A11.815 11.815 0 0012.05 0C5.495 0 .16 5.335.157 11.892c0 2.096.547 4.142 1.588 5.945L.057 24l6.305-1.654a11.882 11.882 0 005.683 1.448h.005c6.554 0 11.89-5.335 11.893-11.893a11.821 11.821 0 00-3.48-8.413z"/></svg>
                Get SafetyLink Now
              </a>
              <a href="#technology" className="btn-out">See How It Works →</a>
            </div>"""
new_hero_btns = """            <div className="hero-btns" style={{ display: 'flex', gap: '16px', flexWrap: 'wrap', justifyContent: 'center' }}>
              <button onClick={onRegisterUser} className="btn-wa" style={{background: '#1e293b', color: '#fff', border: 'none', cursor: 'pointer', padding: '14px 24px', borderRadius: '30px', fontWeight: 'bold', fontSize: '15px', display: 'flex', alignItems: 'center', justifyContent: 'center'}}>Register Individual</button>
              <button onClick={onRegisterOrg} className="btn-wa" style={{background: 'var(--green)', color: '#fff', border: 'none', cursor: 'pointer', padding: '14px 24px', borderRadius: '30px', fontWeight: 'bold', fontSize: '15px', display: 'flex', alignItems: 'center', justifyContent: 'center'}}>Start your 14-day trial</button>
            </div>"""
content = content.replace(old_hero_btns, new_hero_btns)

with open(file_path, "w") as f:
    f.write(content)

File: .restored_features_backup/deep_archive/update-home.js
import fs from 'fs';

const path = 'src/components/landing/Home.tsx';
let content = fs.readFileSync(path, 'utf8');

// Add state for releases
if (!content.includes('const [releases, setReleases]')) {
  content = content.replace(
    'const [mobileMenuOpen, setMobileMenuOpen] = useState(false);',
    'const [mobileMenuOpen, setMobileMenuOpen] = useState(false);\n  const [releases, setReleases] = useState<any[]>([]);\n  const [latestApkUrl, setLatestApkUrl] = useState<string>("");\n  const [latestExeUrl, setLatestExeUrl] = useState<string>("");\n'
  );
}

// Add fetch in useEffect
if (!content.includes('fetch(\'https://api.github.com/repos/Charesmagna/SafetyLink-Core/releases\')')) {
  const fetchEffect = `
  useEffect(() => {
    const fetchReleases = async () => {
      try {
        const res = await fetch('https://api.github.com/repos/Charesmagna/SafetyLink-Core/releases');
        if (res.ok) {
          const data = await res.json();
          setReleases(data);
          if (data.length > 0) {
            const latest = data[0];
            const apkAsset = latest.assets?.find((a: any) => a.name.endsWith('.apk'));
            const exeAsset = latest.assets?.find((a: any) => a.name.endsWith('.exe'));
            if (apkAsset) setLatestApkUrl(apkAsset.browser_download_url);
            if (exeAsset) setLatestExeUrl(exeAsset.browser_download_url);
          }
        }
      } catch (e) {
        console.error('Failed to fetch releases', e);
      }
    };
    fetchReleases();
  }, []);
`;
  content = content.replace(
    'const tourIntervalRef = useRef<ReturnType<typeof setInterval> | null>(null);',
    'const tourIntervalRef = useRef<ReturnType<typeof setInterval> | null>(null);\n' + fetchEffect
  );
}

// Update the download section
const downloadSectionOld = `<div className="dl-grid">
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787313194/Code_Generated_Image_1.png" alt="Android App Preview"/>
              <div className="dl-title">Android APK</div>
              <div className="dl-sub">Minimum Android 8.0. Bluetooth LE required for iTAG functionality.</div>
              <a href="https://wa.me/27739441222?text=I+want+the+SafetyLink+APK" target="_blank" rel="noreferrer" className="dl-btn-link apk">Request on WhatsApp</a>
            </div>
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787313199/Gemini_Generated_Image_.png" alt="Windows Command Deck"/>
              <div className="dl-title">Windows EXE</div>
              <div className="dl-sub">SafetyLink Command Deck desktop app. Requires SL-ORG-XXXX access code.</div>
              <a href="https://wa.me/27739441222?text=I+want+the+SafetyLink+EXE" target="_blank" rel="noreferrer" className="dl-btn-link exe">Request on WhatsApp</a>
            </div>
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787310049/Polish_20260620_014530309.jpg" alt="SafetyLink PWA"/>
              <div className="dl-title">PWA</div>
              <div className="dl-sub">Access directly from your browser. Tap Add to Home Screen. Full offline capability once installed.</div>
              <a href="https://safetylink.online" target="_blank" rel="noreferrer" className="dl-btn-link pwa">Open PWA</a>
            </div>
          </div>`;

const downloadSectionNew = `<div className="dl-grid">
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787313194/Code_Generated_Image_1.png" alt="Android App Preview"/>
              <div className="dl-title">Android APK</div>
              <div className="dl-sub">Minimum Android 8.0. Bluetooth LE required for iTAG functionality.</div>
              <a href={latestApkUrl || "https://github.com/Charesmagna/SafetyLink-Core/releases/latest"} target="_blank" rel="noreferrer" className="dl-btn-link apk">{latestApkUrl ? 'Download Latest APK' : 'View Releases'}</a>
            </div>
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787313199/Gemini_Generated_Image_.png" alt="Windows Command Deck"/>
              <div className="dl-title">Windows EXE</div>
              <div className="dl-sub">SafetyLink Command Deck desktop app. Requires SL-ORG-XXXX access code.</div>
              <a href={latestExeUrl || "https://github.com/Charesmagna/SafetyLink-Core/releases/latest"} target="_blank" rel="noreferrer" className="dl-btn-link exe">{latestExeUrl ? 'Download Latest EXE' : 'View Releases'}</a>
            </div>
            <div className="dl-card">
              <img src="https://res.cloudinary.com/qcp4fx2v/image/upload/f_auto,q_auto/v1787310049/Polish_20260620_014530309.jpg" alt="SafetyLink PWA"/>
              <div className="dl-title">PWA</div>
              <div className="dl-sub">Access directly from your browser. Tap Add to Home Screen. Full offline capability once installed.</div>
              <a href="https://safetylink.online" target="_blank" rel="noreferrer" className="dl-btn-link pwa">Open PWA</a>
            </div>
          </div>
          
          {releases.length > 0 && (
            <div style={{ marginTop: '4rem', background: '#f8fafc', border: '1px solid #e2e8f0', borderRadius: '1rem', padding: '2rem', maxWidth: '800px', margin: '4rem auto 0 auto' }}>
              <h3 style={{ fontSize: '1.25rem', fontWeight: 'bold', color: '#0f172a', marginBottom: '1.5rem', borderBottom: '1px solid #e2e8f0', paddingBottom: '1rem', fontFamily: 'monospace' }}>Release History</h3>
              <div style={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
                {releases.map((release: any) => (
                  <div key={release.id} style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', alignItems: 'center', gap: '1rem', background: 'white', padding: '1rem', borderRadius: '0.75rem', border: '1px solid #f1f5f9', boxShadow: '0 1px 2px rgba(0,0,0,0.05)' }}>
                    <div>
                      <div style={{ display: 'flex', alignItems: 'center', gap: '0.75rem' }}>
                        <span style={{ fontWeight: 'bold', color: '#1e293b', fontSize: '1.125rem' }}>{release.tag_name}</span>
                        {release.prerelease && <span style={{ background: '#fef3c7', color: '#92400e', fontSize: '0.625rem', textTransform: 'uppercase', fontWeight: 'bold', padding: '0.125rem 0.5rem', borderRadius: '9999px' }}>Pre-release</span>}
                      </div>
                      <p style={{ fontSize: '0.875rem', color: '#64748b', marginTop: '0.25rem' }}>{new Date(release.published_at).toLocaleDateString()} - {release.name}</p>
                    </div>
                    <div style={{ display: 'flex', alignItems: 'center', gap: '0.5rem' }}>
                      {release.assets?.map((a: any) => (
                        <a key={a.id} href={a.browser_download_url} target="_blank" rel="noreferrer" style={{ padding: '0.5rem 0.75rem', background: '#f1f5f9', color: '#475569', fontSize: '0.75rem', fontWeight: 'bold', borderRadius: '0.5rem', textDecoration: 'none' }}>
                          {a.name.endsWith('.apk') ? 'APK' : a.name.endsWith('.exe') ? 'EXE' : 'DL'}
                        </a>
                      ))}
                      <a href={release.html_url} target="_blank" rel="noreferrer" style={{ padding: '0.5rem 1rem', background: '#0f172a', color: 'white', fontSize: '0.75rem', fontWeight: 'bold', borderRadius: '0.5rem', textDecoration: 'none' }}>
                        Notes
                      </a>
                    </div>
                  </div>
                ))}
              </div>
            </div>
          )}
`;

content = content.replace(downloadSectionOld, downloadSectionNew);
fs.writeFileSync(path, content, 'utf8');
